blockchain/utxobackend.go

```diff
@@ -171,72 +305,160 @@ func removeRegressionDB(net wire.CurrencyNet, dbPath string) error {
 	return nil
 }
 
+// fileExists reports whether the named file or directory exists.
+func fileExists(name string) bool {
+	if _, err := os.Stat(name); err != nil {
+		if os.IsNotExist(err) {
+			return false
+		}
+	}
+	return true
+}
+
 // LoadUtxoDB loads (or creates when needed) the UTXO database and returns a
 // handle to it.  It also contains additional logic such as ensuring the
 // regression test database is clean when in regression test mode.
-func LoadUtxoDB(params *chaincfg.Params, dataDir string) (database.DB, error) {
-	// Set the database path based on the data directory and database name.
+func LoadUtxoDB(ctx context.Context, params *chaincfg.Params, dataDir string) (*leveldb.DB, error) {
+	// Set the database path based on the data directory and UTXO database name.
 	dbPath := filepath.Join(dataDir, utxoDbName)
 
 	// The regression test is special in that it needs a clean database for each
 	// run, so remove it now if it already exists.
 	removeRegressionDB(params.Net, dbPath)
```

Comment:
I don't think this should check the error personally, because it doesn't really matter if it isn't removed.  It's just a "nice to have" cleanup and shouldn't prevent normal startup if it can't remove it.

The one in the main code doesn't check the error for the same reason.  For some context, I've done some testing before with removable drives and such and had it fail to startup due to this before which is why it doesn't check the error there.

Reply:
Fair enough.  Also not in scope for this PR, but this suggests that the function shouldn't have an error return imo.

Reply:
I figured it’d be fine since it only impacts `regnet`, but I can go ahead and revert that bit since that is how the main database path handles it anyway.

Reply:
Updated and changed to `_ = removeRegressionDB(params.Net, dbPath)` to make it clear that it's intentionally ignoring the error.

Reply:
@chappjc Yeah, that makes sense.  I typically like to return errors from funcs that might end up in tests, but given the scope here, that seems fair.